File: ARCADE/Intro/Level 7_2: depositProfit.cs
    using System;
     
    public class Test
    {
    	public static int depositProfit(int deposit, int rate, int threshold){
            double a = Convert.ToDouble(deposit);
            int year = 0;
            while (a < threshold){
                a = a * rate / 100 + a;
                year++;
            }
            return year;
        }
    	public static void Main()
    	{
    		int deposit = 100, rate = 20, threshold = 170;
    		Console.WriteLine(depositProfit(deposit, rate, threshold));
    	}
    }

File: ARCADE/Intro/Level 7_2: depositProfit.pas
program depositProfit;
 
function depositProfit(deposit: integer; rate: integer; threshold: integer): integer;
var rs : integer;
    a : double;
begin
    a := deposit; 
    rs := 0;
    while(a < threshold) do
    begin
        a := a + (a * rate / 100);
        inc(rs);
    end;
 
    depositProfit := rs;
end;
 
var deposit, rate, threshold : integer;
begin
	deposit := 100; rate := 20; threshold := 170;
	writeln(depositProfit(deposit, rate, threshold));
end.
